packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2896,6 +2901,23 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin, Resto
   }
 }
 
+class _DismissDrawerAction extends DismissAction {
+  _DismissDrawerAction(this.context);
+
+  final BuildContext context;
+
+  @override
+  bool isEnabled(DismissIntent intent) {
+    return Scaffold.of(context).isDrawerOpen || Scaffold.of(context).isEndDrawerOpen;
+  }
+
+  @override
+  void invoke(DismissIntent intent) {
+    Scaffold.of(context).closeDrawer();
+    Scaffold.of(context).closeEndDrawer();
+  }
```

Comment:
This is added unconditionally since these functions already check if there is a drawer and if it is open. 
Ex:

https://github.com/flutter/flutter/blob/526c33fe30e5475abc7c65650274e53b3627de5b/packages/flutter/lib/src/material/scaffold.dart#L2156-L2160